src/Avalonia.Controls/WindowBase.cs

```diff
@@ -284,6 +283,11 @@ protected override Size MeasureCore(Size availableSize)
 
             var constraint = LayoutHelper.ApplyLayoutConstraints(this, availableSize);
 
+            if (this is Styling.IContainer container)
```

Comment:
Note that this condition is currently always true since WindowBase is a ContentControl

Reply:
Also, why is it in WindowBase and not in TopLevel?

Reply:
TopLevel doesn't have any special MeasureCore implemention, relying on the base implementation in Layoutable and the override in WindowBase

Reply:
1) TopLevels should **always** act as Width+Height containers for satisfying media queries
2) TopLevels should **never** be affected by the new Measure logic in Layoutable

Reply:
1) TopLevels should **always** act as Width+Height containers for satisfying media queries
2) TopLevels should **never** be affected by the new Measure logic in Layoutable
3) Manually setting ContainerType/ContainerName on a TopLevel is invalid and should trigger an exception